intellij/src/saros/intellij/negotiation/ModuleConfigurationInitializer.java

```diff
@@ -81,43 +95,163 @@ public void enqueueModuleConfigurationChange(
   private void applyModuleConfiguration(@NotNull IProject wrappedModule) {
     Module module = wrappedModule.adaptTo(IntelliJProjectImpl.class).getModule();
 
-    Map<String, String> configurationOptions = queuedModuleOptions.remove(module);
+    ModuleConfiguration moduleConfiguration = queuedModuleOptions.remove(module);
 
-    if (configurationOptions == null) {
+    if (moduleConfiguration == null) {
       log.debug("Skipped module root model adjustment as not options were found - " + module);
 
       return;
     }
 
-    ModuleRootModificationUtil.updateModel(
-        module,
-        modifiableRootModel -> applyModuleConfiguration(modifiableRootModel, configurationOptions));
+    if (!moduleConfiguration.isExistingModule()) {
```

Comment:
code can be changed so it is not nested so deep, see below

Reply:
Done.